x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -208,47 +218,98 @@ static void getMemoryAndProcessors(
+                        extraProcessors += numMissingProcessors - numExistingProcessorsToBeUsed;
+                    }
+                    numberOfAvailableProcessors -= numExistingProcessorsToBeUsed;
                 }
             } else if (assignment.getNodeRoutingTable().values().stream().allMatch(r -> r.getState().consumesMemory() == false)) {
-                // Ignore states that don't consume memory, for example all allocations are failed
+                // Ignore states that don't consume memory, for example all allocations are failed or stopped
                 continue;
+            } else if (AssignmentState.STARTED.equals(assignment.getAssignmentState()) && numMissingAllocations > 0) {
+                // this state occurs when we have a partial allocation, but not fully allocated
+
+                long numMemoryBytesNeeded = assignment.getTaskParams().estimateMemoryUsageBytes();
+
+                if (numberOfAvailableProcessors < numMissingProcessors) {
+                    // we don't have enough processors or memory to start the new allocations, we need to scale up
+
+                    if (numberOfAvailableProcessors < numMissingProcessors) {
+                        numberOfAvailableProcessors -= numExistingProcessorsToBeUsed;
+                        extraProcessors += numMissingAllocations * numberOfThreadsPerAllocation - numExistingProcessorsToBeUsed;
+                        extraSingleNodeProcessors = Math.max(extraSingleNodeProcessors, numberOfThreadsPerAllocation);
+                    }
+
+                    final int extraProcessorsNeededForAssignment = numMissingAllocations * numberOfThreadsPerAllocation
+                        - numExistingProcessorsToBeUsed;
+                    logger.warn(
+                        () -> format(
+                            "trained model [%s] assigned to [%s], waiting for [%d] allocations to start; waiting on [%d] extra processors "
+                                + "to become available",
+                            modelAssignment.getKey(),
+                            Strings.arrayToCommaDelimitedString(modelAssignment.getValue().getStartedNodes()),
+                            numMissingAllocations,
+                            extraProcessorsNeededForAssignment
+                        )
+                    );
+                } else {
+                    logger.info(
+                        () -> format(
+                            "trained model [%s] assigned to [%s], waiting for [%d] allocations to start",
+                            modelAssignment.getKey(),
+                            Strings.arrayToCommaDelimitedString(modelAssignment.getValue().getStartedNodes()),
+                            numMissingAllocations
+                        )
+                    );
+                    numberOfAvailableProcessors -= numExistingProcessorsToBeUsed;
```

Comment:
Shouldn't this be in both branches of the conditional rather than just the else?